docs/component-api.md

```diff
@@ -0,0 +1,287 @@
+  onclick="toggle()">` when a button is required, an `<a>` tag when a link is
+  required, and so on.
+- **Clarity over brevity.** Developers should be able to understand what's going
+  on with markup at a glance. Avoid cryptic abbreviations and nicknames, add
+  proper indenting & spacing, and use clear comments.
+- **Native HTML elements** (e.g. `<input>`, `<select>`) should be preferred over
+  custom elements whenever possible. Native elements provide a slew of
+  functionality and accessibility best practices out of the box.
+
+## CSS
+
+### CSS design principles
+
+- **Portable** - The CSS architecture uses CSS classes for styling (versus CSS-in-JS
+  tooling) in order to ensure the CSS is portable across frameworks and web
+  technologies.
+- **Clarity over brevity** - CSS class naming conventions are verbose, but they
+  deliver clarity, legibility, and reslience in exchange.
+- **Modular** - Component styles are fully modular in order to keep things tightly
+  scoped and to avoid unintended style bleeding.
+- **Limit chaining and multiple selectors** - Chaining and descendant selectors
+  should be avoided wherever possible in order to keep CSS as DOM-independent
+  and modular as possible.
+
+### CSS conventions
+
+OUI follows a [BEM](http://getbem.com/introduction/)-like syntax, extending it
+further to follow more of
+[BEMIT](http://csswizardry.com/2015/08/bemit-taking-the-bem-naming-convention-a-step-further/)-like
+conventions.
+
+
+#### BEM syntax
+
+BEM stands for "Block Element Modifier". Here's a breakdown of what that means:
+
+- **Block** is the primary component block (e.g. .oui-c-button)
+- **Element** is a child of the primary block (e.g. .oui-c-button__text)
+- **Modifier** is a variation of a component style (e.g. .oui-c-button--secondary)
+
+OUI extends BEM's conventions to create even more explicit, encapsulated class
+names.
+
+#### Global namespace
+
+OUI uses a global namespace of `oui-` prefix to all styles that come from the
+design system. This is done to:
+
+  - Avoid naming collisions with code coming from other sources
+  - Clarify the source of the code, distinguishing OUI markup/styles from code
+    coming from other codebases and libraries.
+
+### Class prefixes
+
+In addition to the global `oui-` namespace, OUI uses class prefixes to provide
+additional clarity to the job a given class plays. OUI uses the following class
+prefix conventions:
+
+- **c** - for UI components, such as `.oui-c-button`
+- **l** - for layout-specific component styles, such as `.oui-l-container`
+- **u** - for utilities, such as `.oui-u-margin-bottom-none`
+- **is** - and has- for specific states, such as `.oui-is-active`
+
+### Logical properties
+
+Use logical properties (`inset-block-start`, `margin-inline`) over physical
+properties (`top`, `margin-left`), to allow the components to be used in
+different writing systems.
+
+### Design tokens
+
+TODO: Define guidelines for design tokens
```

Comment:
@bradfrost Would you be able to suggest guidelines here?

Reply:
@jamesnw Yep, I'll likely handle this as a separate issue and file. Historically, we've had a `THEMING.md` file that gets into all of that architecture.